triton-client/src/store/thunks.ts

```diff
@@ -153,12 +153,24 @@ export const createDownloadRequest = (projectId: ExternalProjectID, datasetID: n
 		}
 	}
 
-export const fetchConstants = () => async (dispatch: AppDispatch, getState: () => RootState) => {
-	try {
-		const constants = await apiTriton.getConstants()
-		dispatch(ConstantsStateActions.setConstants(constants))
-	} catch (err) {
-		dispatch(ConstantsStateActions.setError(convertToSerializedError(err)))
-		throw err
-	}
-}
+  export const fetchConstants = () => async (dispatch: AppDispatch, getState: () => RootState) => {
+    try {
+      const constants = await apiTriton.getConstants()
+      dispatch(ConstantsStateActions.setConstants(constants))
+    } catch (err) {
+      dispatch(ConstantsStateActions.setError(convertToSerializedError(err)))
+      throw err
+    }
+  }
+
+  export const deleteDownloadRequest = (projectId: ExternalProjectID, datasetID: number) =>
+    async (dispatch: AppDispatch, getState: () => RootState) => {
+      try {
+        const response = await apiTriton.deleteDownloadRequest(datasetID)
+        // console.debug(`Loaded datasets succesfully: ${JSON.stringify(datasets)}`)
+        dispatch(DatasetsStateActions.setDownloadRequest(response))
+      } catch (err: any) {
+        dispatch(DatasetsStateActions.setError({ projectId, error: convertToSerializedError(err) }))
+        throw err
+      }
```

Comment:
I'm starting to wish we bothered to print error instead of just serializing error into the store.

Reply:
Not sure what you mean here...

Reply:
Right now we are just calling setError in thunks whenever any error happened. But it would be nice if we also printed to console the error that occurred. Although, we do already get error from requests failing anyway.

Reply:
I am not ever sure why store errors in state if we don't do anything about it ...
Maybe we can build a error console !

Reply:
Yeah it should be another PR. It was more like venting than actually expecting you to do something.